tests/particles/exchange_ghosts_periodic_01.cc

```diff
@@ -0,0 +1,135 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// like particle_handler_06, but tests that the particle handlers function
+// n_locally_owned_particles only counts locally owned particles.
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+#include "../tests.h"
+
+template <int dim, int spacedim>
+void
+test()
+{
+  {
+    parallel::distributed::Triangulation<dim, spacedim> tr(MPI_COMM_WORLD);
+
+    // Create a subdivided hyper rectangle with divisions along the x axis
+    // Colorize is set to true to enable setting one set of periodic boundary
+    // conditions along id 0 and 1
+    std::vector<unsigned int> repetitions({6, 1});
+    if (dim == 3)
+      repetitions.push_back(1);
+    Point<spacedim> left;
+    Point<spacedim> right;
+    for (int i = 0; i < spacedim; ++i)
+      right[i] = 1;
+
+    GridGenerator::subdivided_hyper_rectangle(
+      tr, repetitions, left, right, true);
+    MappingQ<dim, spacedim> mapping(1);
+
+
+    // Generate periodic boundary conditions
+    std::vector<GridTools::PeriodicFacePair<
+      typename Triangulation<dim, spacedim>::cell_iterator>>
+      periodicity_vector;
+    GridTools::collect_periodic_faces(tr, 0, 1, 0, periodicity_vector);
+    tr.add_periodicity(periodicity_vector);
+
+    // both processes create a particle handler with a particle in a
+    // position that is in a ghost cell to the other process connected by a
+    // periodic boundary
+    Particles::ParticleHandler<dim, spacedim> particle_handler(tr, mapping);
+
+    Point<spacedim> position;
+    Point<dim>      reference_position;
+
+    if (Utilities::MPI::this_mpi_process(tr.get_communicator()) == 0)
+      position(0) = 0.001;
```

Comment:
Just for consistency 

```suggestion
      position[0] = 0.001;
```